library/cwm/src/lib/cwm/widget.rb

```diff
@@ -0,0 +1,631 @@
+  # The {#label} method is mandatory.
+  #
+  # @see ComboBox for child example
+  class SelectionBox < AbstractWidget
+    self.widget_type = :selection_box
+
+    include ItemsSelection
+    abstract_method :label
+
+    # @return [String] ID of the selected item
+    def value
+      Yast::UI.QueryWidget(Id(widget_id), :CurrentItem)
+    end
+
+    # @param val [String] ID of the selected item
+    def value=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :CurrentItem, val)
+    end
+  end
+
+  # A multi-selection box to select more values.
+  # The {#label} method is mandatory.
+  #
+  # @see {ComboBox} for child example
+  class MultiSelectionBox < AbstractWidget
+    self.widget_type = :multi_selection_box
+
+    include ItemsSelection
+    abstract_method :label
+
+    # @return [Array<String>] return IDs of selected items
+    def value
+      Yast::UI.QueryWidget(Id(widget_id), :SelectedItems)
+    end
+
+    # @param val [Array<String>] IDs of newly selected items
+    def value=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :SelectedItems, val)
+    end
+  end
+
+  # An integer field widget.
+  # The {#label} method is mandatory.
+  # It supports optional {#minimum} and {#maximum} methods
+  # for limiting the range.
+  # See {#cwm_definition} method for minimum and maximum example
+  #
+  # @see InputField for example of child
+  class IntField < AbstractWidget
+    self.widget_type = :intfield
+
+    include ValueBasedWidget
+    abstract_method :label
+
+    # @!method minimum
+    #   @return [Fixnum] limited by C signed int range (-2**30 to 2**31-1).
+
+    # @!method maximum
+    #   @return [Fixnum] limited by C signed int range (-2**30 to 2**31-1).
+
+    # The definition for IntField additionally supports
+    # `minimum` and `maximum` methods.
+    # @example minimum and maximum methods
+    #   def minimum
+    #     50
+    #   end
+    #
+    #   def maximum
+    #     200
+    #   end
+    def cwm_definition
+      res = {}
+
+      res["minimum"] = minimum if respond_to?(:minimum)
+      res["maximum"] = maximum if respond_to?(:maximum)
+
+      super.merge(res)
+    end
+  end
+
+  # A selection of a value via radio buttons.
+  # The {#label} method is mandatory.
+  #
+  # @see {ComboBox} for child example
+  class RadioButtons < AbstractWidget
+    self.widget_type = :radio_buttons
+
+    include ItemsSelection
+    abstract_method :label
+
+    def value
+      Yast::UI.QueryWidget(Id(widget_id), :CurrentButton)
+    end
+
+    def value=(val)
+      Yast::UI.ChangeWidget(Id(widget_id), :CurrentButton, val)
+    end
+  end
+
+  # Widget representing button.
+  #
+  # @example push button widget child
+  #   class MyEvilWidget < CWM::PushButton
+  #     def label
+  #       _("Win the lottery by clicking this.")
+  #     end
+  #
+  #     def handle
+  #       Virus.install
+  #       nil
+  #     end
+  #   end
+  class PushButton < AbstractWidget
+    self.widget_type = :push_button
```

Comment:
If `abstract_method :label` is omitted on purpose, please add a comment why. It seems like a mistake to me.